qiskit/result/utils.py

```diff
@@ -118,10 +127,54 @@ def _adjust_creg_sizes(creg_sizes, indices):
     return new_creg_sizes
 
 
+def marginal_distribution(
+    counts: dict, indices: Optional[List[int]] = None, format_marginal: bool = False
+) -> Dict[str, int]:
+    """Marginalize counts from an experiment over some indices of interest.
+
+    Unlike :func:`~.marginal_counts` this function respects the order of
+    the input ``indices``. If the input ``indices`` list is specified, the order
+    the bit indices will be the output order of the bitstrings
+    in the marginalized output.
+
+    Args:
+        counts: result to be marginalized
+        indices: The bit positions of interest
+            to marginalize over. If ``None`` (default), do not marginalize at all.
+        format_marginal: Default: False. If True, takes the output of
+            marginalize and formats it with placeholders between cregs and
+            for non-indices.
+    Returns:
+        dict(str, int): A marginalized dictionary
+    Raises:
+        QiskitError: If any value in ``indices`` is invalid or the ``counts`` dict
+        is invalid.
+    """
+    num_clbits = len(max(counts.keys()).replace(" ", ""))
```

Comment:
Is this inferring the number of classical bit based on the length of the bit string? For qubit-based systems this is fine but for systems that have more than 10 states then this would not work. Is there a way to accommodate this or do we simply assume we do not cover such cases?

Reply:
We can pass in the number of classical bits as a parameter, or maybe have a provision for a callback function that if specified will be passed be passed the largest key and return the size. There is flexibility in the implementation here because it's a new function, if we need to adjust it to support different kinds of systems. I was only thinking about qubit systems when I wrote this but if you have a suggestion on how you'd like to support that now is the time to adjust this.